lib/lndclient/LndClient.ts

```diff
@@ -693,26 +714,39 @@ class LndClient extends SwapClient {
     amount,
     finalCltvDelta,
     cltvLimit,
+    payReq,
   }: {
     rHash: string;
     destination: string;
     amount: number;
     finalCltvDelta: number;
     cltvLimit?: number;
+    payReq?: string;
   }): lndrouter.SendPaymentRequest => {
     const request = new lndrouter.SendPaymentRequest();
-    request.setPaymentHash(Buffer.from(rHash, 'hex'));
-    request.setDest(Buffer.from(destination, 'hex'));
-    request.setAmt(amount);
-    request.setFinalCltvDelta(finalCltvDelta);
-    request.setTimeoutSeconds(MAX_PAYMENT_TIME / 1000);
-    const fee = Math.floor(MAX_FEE_RATIO * request.getAmt());
-    request.setFeeLimitSat(fee);
+    if (payReq) {
+      request.setPaymentRequest(payReq);
+      request.setMaxParts(LndClient.MAX_PARTS);
+    } else {
+      // TODO: as of lnd 0.12.0, this won't work as PaymentAddr will be required and the only way to specify it will be with the pay req
```

Comment:
I don't know how I feel about keeping this code. Maybe in the first release after LND v0.12.0 we should remove it?

Reply:
It's basically there for backwards compatibility with older lnd and older xud versions, it's ignored by other peers using any version of xud that includes the commits from this PR. I figure removing it after we've confirmed that xud works with lnd 0.12+ would be fine, in my one quick test using 0.12 for the simtests caused them to break so there's some work/investigation to be done there.